core/kotlinx-coroutines-test/src/internal/ThreadSafeHeap.kt

```diff
@@ -0,0 +1,153 @@
+/*
```

Comment:
@elizarov This file is a direct copy of the one in core. However, I'm not sure the best way to avoid code duplication here. Happy to update the import however makes sense.

Reply:
I'm changing this to reuse from the core from internal module.